### DISCUSSION TITLE
split event with > 20 objectIDs

### DISCUSSION THREAD

**Discussion by eunjae-lee** [algolia/instantsearch#4963] [Ideas]
Since this PR https://github.com/algolia/instantsearch.js/pull/4841, an event with more than 20 objectIDs is split into multiple requests. Now that multiple events in a single request is supported in search-insights (https://github.com/algolia/search-insights.js/pull/335), we can update InstantSearch.js as well to send a single request instead of many.

I propose to revert some of the part of the previous work, which is splitting payload and call `instantSearchInstance.sendEventToInsights()` multiple times. For those using `onEvent` hook to get the payload, this can be confusing because it's supposed to be a single callback for them. I think we can split the job within `instantSearchInstance.sendEventToInsigths()` method.

This line
https://github.com/algolia/instantsearch.js/blob/master/src/middlewares/createInsightsMiddleware.ts#L155:L155

can become something like this:
```js
insightsClient('sendEvents', chunk(event.payload));
```

What do you think?

**Comment by Haroenv** [comment]
interesting, but won't that break people on older versions of search-insights?

**Reply by eunjae-lee** [reply]
yeah if we decide to this, then we should either do this in the next major version of InstantSearch.js, or put a big banner in the changelog?

**Reply by Haroenv** [reply]
yes, in a major version I see it as possible, but probably not before
